perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/UsersManagerBlImpl.java

```diff
@@ -547,14 +549,40 @@ public void deleteUser(PerunSession sess, User user, boolean forceDelete) throws
 		}
 
 
-		// Delete assigned attributes
-		// Users one
+		// Delete or anonymize assigned attributes
 		try {
-			getPerunBl().getAttributesManagerBl().removeAllAttributes(sess, user);
 			// User-Facilities one
 			getPerunBl().getAttributesManagerBl().removeAllUserFacilityAttributes(sess, user);
-		} catch(WrongAttributeValueException | WrongReferenceAttributeValueException ex) {
-			//All members are deleted => there are no required attribute => all atributes can be removed
+
+			// Users one
+			if (anonymizeInstead) {
+				List<String> attrNames = BeansUtils.getCoreConfig().getAttributesToAnonymize();
+				List<Attribute> userAttributes = getPerunBl().getAttributesManagerBl().getAttributes(sess, user);
+				for (Attribute attribute : userAttributes) {
+					// Skip core attributes
+					if (getPerunBl().getAttributesManager().isCoreAttribute(sess, attribute)) {
+						continue;
+					}
```

Comment:
I think you should also skip virtual attributes.

Reply:
Fixed.